client/src/components/overview/ProductOverview.jsx

```diff
@@ -88,25 +121,35 @@ class ProductOverview extends React.Component {
     }
   }
 
-  updateDefaultStyle() {
+  updateDefaultStyle(styles) {
     console.log('Updating default style');
-    const defaultStyle = this.getDefaultStyle();
+    const defaultStyle = this.getDefaultStyle(styles);
     this.updateState(defaultStyle);
   }
 
-  getDefaultStyle() {
-    for (const style of this.styles) {
+  updateRatingsProperties(ratings) {
+    console.log(`Product Ratings: ${JSON.stringify(ratings)}`);
+    const numberOfReviews = this.getNumberOfReviews(ratings);
+    this.setState({
+      numberOfReviews: numberOfReviews,
+      averageRating: this.getAverageRating(ratings, numberOfReviews)
+    });
+  }
+
+  // TODO: Begin model refactor section
+  getDefaultStyle(styles) {
+    for (const style of styles) {
       if (style['default?']) {
         return style;
       }
     }
     console.log('No default found! Using first style...');
-    return this.styles.length > 0 ? this.styles[0] : null;
+    return styles.length > 0 ? styles[0] : null;
   }
 
-  getStyleById(id) {
+  getStyleById(id, styles) {
```

Comment:
`checkCache` and `updateCache` use a Map object to performs this sort of operation in constant time.

Reply:
Yep, I saw that. A later refactor sweep to do after some other optimizations. For now the scope of this PR was just to stop making REST calls isolated from the rest of the app. I see more refactoring to do in later PRs.